src/org/labkey/test/components/ui/DeleteConfirmationDialog.java

```diff
@@ -59,8 +74,7 @@ public ConfirmPage confirmDelete()
 
     public ConfirmPage confirmDelete(Integer waitSeconds)
     {
-        this.dismiss("Yes, Delete", waitSeconds);
-        return _confirmPageSupplier.get();
+        return _confirmPageSupplier.apply(() -> this.dismiss("Yes, Delete", waitSeconds));
```

Comment:
One day when we have a moment together I'd like to talk to you about how this works